### ISSUE TITLE
Virtual Network Gateway Doc/Schema Mismatch

### ISSUE THREAD

**Issue by marstr** [Azure/azure-rest-api-specs#882]
As seen in Azure/azure-sdk-for-go#261 created by @stack72:

there are differences between the SDK and the documentation for the virtual network gateway. Currently, the [docs](https://msdn.microsoft.com/en-us/library/mt130667.aspx) suggest that we pass the following style of request to the API:

```
{
    "name": "myvirtualgateway1",
    "location": "West US",
    "tags": {
        "key1": "value1"
    },
    "properties": {
        "gatewayType": "DynamicRouting",
        "gatewaySize": "Default",
        "bgpEnabled": True,
        "vpnClientAddressPool": ["{vpnClientAddressPoolPrefix}"],
        "defaultSites": ["mysite1"]
    }
}
```
1. Documentation suggests that GatewayType should be `StaticRouting` or `DynamicRouting`. When looking at the SDK, all I can see for GatewayType is as follows:

`network.VirtualNetworkGatewayType`

this is an Enum for `ExpressRoute` or `Vpn`. So the values in the docs wouldn't line up with any of those Enum values
2. The documentation says that the gatewaySize is a string. When using the SDK, I cannot see that:

```
type VirtualNetworkGatewayPropertiesFormat struct {
    IPConfigurations       *[]VirtualNetworkGatewayIPConfiguration `json:"ipConfigurations,omitempty"`
    GatewayType            VirtualNetworkGatewayType               `json:"gatewayType,omitempty"`
    VpnType                VpnType                                 `json:"vpnType,omitempty"`
    EnableBgp              *bool                                   `json:"enableBgp,omitempty"`
    GatewayDefaultSite     *SubResource                            `json:"gatewayDefaultSite,omitempty"`
    Sku                    *VirtualNetworkGatewaySku               `json:"sku,omitempty"`
    VpnClientConfiguration *VpnClientConfiguration                 `json:"vpnClientConfiguration,omitempty"`
    ResourceGUID           *string                                 `json:"resourceGuid,omitempty"`
    ProvisioningState      *string                                 `json:"provisioningState,omitempty"`
}
```

the size is not in the SDK to send as part of the request
1. The docs also say that the Default Site is a string but the SDK suggests that the `GatewayDefaultSite` is a SubResource. What is it a SubResource of?
2. The docs suggest that the `vpnClientAddressPool` is a string for a single address. Whereas the SDK requires me to construct the following:

```
vpnConfiguration := network.VpnClientConfiguration{
        VpnClientAddressPool: &network.AddressSpace{},
    }
```

Where AddressSpace is a *[]string. Which is the correct source of truth?

So to summarise, the only part of the resource that aligns between docs and SDK is `enable_bgp` (which is a bool), `name` (string) and `location` (string)

Paul

**Comment by amarzavery** [comment]
@lamchester - Can you please take a look at this issue posted by one of the customers using the Go sdk? It seems there is a mismatch between what is represented in the swagger and what is happening on the wire.

> On that note, to ease customer pain, we have started asking service teams to provide examples of requests and responses for every operation in the swagger spec.

If you can provide (minimum [just the required properties/parameters for making a request] and maximum [full blown request with all the required and optional properties/parameters]) examples  for request and response using the extension **"x-ms-examples"** then it will,
- help us **validate the examples against the swagger spec** and catch errors early on and 
- it will also form the **basis for customer facing REST API documentation** on docs.microsoft.com website.

You can more info about the extension and an example swagger spec over [here](https://github.com/Azure/adx-documentation-pr/blob/master/tools/tools.md#swagger-model-validation).

**Comment by lamchester** [comment]
The doc is wrong and (badly) out of date. What's being generated by swagger is correct, and the swagger spec is the source of truth here. 

This is really not what the experience should be and I apologize for that. I will look into both getting the doc updated, and providing an example as suggested. Thanks for bringing this up!

**Comment by lamchester** [comment]
@yushwang can you take a look at the docs?

**Comment by yushwang** [comment]
@amarzavery Thanks for the feedback, and as you pointed out, the MSDN docs are completely outdated and inaccurate. We are in the process of fixing them, but also in the process of migrating those to a different publication system. The actual migration should be done within these two weeks, but will be whatever is on MSDN due to the lockdown requirements for migration, then we will start fixing all the bugs. Please bear with us a bit more.

Thanks,
Yushun

**Comment by markjbrown** [comment]
We have now published docs directly from our swagger spec so this issue is now fixed. @lamchester  can you please close this issue? CC @liumichelle

**Comment by stack72** [comment]
Thanks for the information @markjbrown 

Closing this as per your recommendation now

**Comment by salameer** [comment]
Closing accroding to @markjbrown comments
